### PR TITLE
Add getAllMealsThatContainString

### PR DESCRIPTION
**Pull request by acrivopoulosdotcom**

### CODE REVIEW

File: Backend/src/main/java/com/example/backend/ShopController.java [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 import java.util.ArrayList;
 import java.util.List;
 
+@RequiredArgsConstructor
 @RestController
 @RequestMapping("/api/meals")
 public class ShopController {
 
-    private final ShopServiceMeals shopServiceMeals = new ShopServiceMeals();
```
Comment (acrivopoulosdotcom) [inline Backend/src/main/java/com/example/backend/ShopController.java:12]: delete

File: Backend/src/main/java/com/example/backend/ShopController.java [hunk 2]
```diff
@@ -27,8 +29,8 @@ public List<Meal> getMealsByArea(@PathVariable String area) {
     }
 
     @GetMapping("/MealsByCategory")
-    public List<Meal> getMealsByCategoy(@PathVariable String category) {
```
Comment (acrivopoulosdotcom) [inline Backend/src/main/java/com/example/backend/ShopController.java:30]: delete

File: Backend/src/main/java/com/example/backend/ShopController.java [hunk 3]
```diff
@@ -27,8 +29,8 @@ public List<Meal> getMealsByArea(@PathVariable String area) {
...
-        return shopServiceMeals.getMealByCategory(category);
```
Comment (acrivopoulosdotcom) [inline Backend/src/main/java/com/example/backend/ShopController.java:31]: delete

**Review by acrivopoulosdotcom — COMMENTED** [review]
first merge backend

**Review by acrivopoulosdotcom — COMMENTED** [review]
merge backend one

**Review by AndyBlank2407 — APPROVED** [review]

**Pull request merged**